### ISSUE TITLE
luaotfload and Math fonts: cvNN feature for superscripts not applied

### ISSUE THREAD

**Issue by dflipo** [latex3/luaotfload#266]
This issue is similar to https://github.com/latex3/luaotfload/issues/227.

For the Math fonts I maintain, I use cvNN features to to toggle Greek variants in Maths, f.i. the cv06 feature makes \phi output mitvarphi (U+1D711) instead of mitphi (U+1D719) the default.  It works fine for base characters, but I recently noticed a problem with the superscripts of these variants (bolder glyphs mitvarphi.st and mitvarphi.sst are available for the ssty feature):  only *one* feature is applied by luaotfload, actually the last called by the font (not the first as I wrote previously).

I reported on SE (https://tex.stackexchange.com/questions/698768/) and after private exchanges with with Mikael Sundquist and Hans Hagen, they came up with a solution (thanks to both of them!) which works for me (see the appended "fontloader-2023-08-19.lua" file).

The following MWE:  *requires  XCharter-Math v0.51*  pushed to CTAN today. In this new version,  feature ssty is applied last (it was applied before all cvNN features in previous versions). Screen captures of the compilation by current LuaLaTeX,  XeLaTeX (correct output) and  LuaLaTeX+patch (correct output) are appended.

`\documentclass{article}
\usepackage{unicode-math}
\usepackage{iftex}
\begin{document}
\setmathfont{XCharter-Math.otf}
\ifluatex LuaLaTeX: \fi
\ifxetex  XeLaTeX:  \fi

\vspace{.5\baselineskip}
\(\phi^{\phi^{\phi}}, v^{v^{v}}\)

\setmathfont{XCharter-Math.otf}[CharacterVariant={6,11}]
\(\phi^{\phi^{\phi}}, v^{v^{v}}\)
\end{document}`

Outputs (screen captures):
![image](https://github.com/latex3/luaotfload/assets/6177282/09c0deea-638b-476e-b6dd-17513da3978e)

![image](https://github.com/latex3/luaotfload/assets/6177282/cd24c60b-0d5c-48b7-97e5-619080ed679b)

After adding file "fontloader-2023-08-19.lua" to the current directory, the output is correct:

![image](https://github.com/latex3/luaotfload/assets/6177282/25f477c2-dc44-4b62-bc87-330f15120412)

[fontloader-2023-08-19.lua.zip](https://github.com/latex3/luaotfload/files/13181132/fontloader-2023-08-19.lua.zip)

**Comment by zauguin** [comment]
@dflipo When you talked to Has Hagen about this, did he mention if he plans to add this change to ConTeXt MKIV soon too?

**Comment by antonis-tsolomitis** [comment]
This bug seems to remain unresolved. The above fix with the extra fontloader file does NOT work with StylisticSets. NewComputerModern is to provide scriptstyle and scriptscriptstyle glyphs for the alternate calligraphic Latin capitals. Let me call those "mathscr". The default is "mathcal" and StylisticSet=1 replaces mathcal with mathscr. This works fine with xelatex. But with lua only the base of an exponent is changed to mathscr. So if you write \mathcal{A}^{\mathcal{A}} you get the base A in mathscr and the exponent A in mathcal.

**Comment by zauguin** [comment]
> This bug seems to remain unresolved. The above fix with the extra fontloader file does NOT work with StylisticSets

When trying the changed fontloader with NewComputerModern's ss01 it does work for me.

**Comment by niruvt** [comment]
> When trying the changed fontloader with NewComputerModern's ss01 it does work for me.

Hello @zauguin,

How does one use the changed fontloader? Without using the fontloader, I provide a MWE:

```
\documentclass[border=1cm]{standalone}
\usepackage{fontspec}
\setmainfont[%
  Renderer          = {Harfbuzz},%
  StylisticSet      = {5},%
  CharacterVariant  = {5},%
  Extension         = {.otf}%
]{NewCM10-Book}

\begin{document}
x͜y\textsuperscript{x͜y}
\end{document}
```

![image](https://github.com/latex3/luaotfload/assets/54938729/6a139da7-9787-4454-90cb-6e70d10f37ee)

My system is up-to-date. Can you reproduce this image with the fontloader?

**Comment by niruvt** [comment]
@zauguin, forgot to mention, for me, the following didn't work:

> After adding file "fontloader-2023-08-19.lua" to the current directory, the output is correct:

**Comment by niruvt** [comment]
Even with XeLaTeX, it fails:

![image](https://github.com/latex3/luaotfload/assets/54938729/2961f710-b914-4394-a8d7-a7e3b4a39c20)

**Comment by zauguin** [comment]
@niruvt This is an different issue. The font you are using has different optical sizes for 10pt and 8pt and the 10pt version does not seem to handle the combining character correctly here.

**Comment by niruvt** [comment]
@zauguin Okay, my bad.

**Comment by niruvt** [comment]
So I and Antonis are discussing a to-be-released feature of NewCM which adds a different shape for number 1 with a character variant. Please use [this](https://github.com/latex3/luaotfload/files/13786552/newcm-unreleased.zip) font for testing (thanks Antonis for allowing sharing the font). Do you think here also the bug is in NewCM? If yes, then both me and Antonis are curious to know how can it be solved? It would be helpful if you can guide.

```
\documentclass[border=1cm]{standalone}
\usepackage{fontspec}
\setmainfont[%
  Numbers                = {OldStyle},%
  CharacterVariant       = {6},%
  Extension              = {.otf}%
]{NewCM10-Book}

\begin{document}
1\textsuperscript{1}
\end{document}
```

![image](https://github.com/latex3/luaotfload/assets/54938729/7a46d0bb-5b02-4b5c-b7e9-0d84a0b64af3)

**Comment by zauguin** [comment]
It's the same as the previous issue: The subscript is smaller and therefore uses a different font. If you only provide a updated NewCM10-Book and not an extended NewCM08-Book you will only affect the bigger one.

**Comment by niruvt** [comment]
@zauguin yes, I figured it out and hence deleted the comment, thanks for the attention!

**Comment by antonis-tsolomitis** [comment]
My system did not have the latest updates. After updating and after placing the above fontloader file in t he same directory NewCM produces correct output. So now the question is when will this be released? CTAN has a file with the same name but it is different than the one above.

**Comment by zauguin** [comment]
> So now the question is when will this be released?

This is now merged into the dev branch and will probably be part of the next release.

**Comment by u-fischer** [comment]
This has been fixed already some time ago.
